tests/unit_tests/components/config.cpp

```diff
@@ -0,0 +1,143 @@
+#include <iostream>
```

Comment:
These tests would greatly benefit from using [test fixtures](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests). The code in the tests is basically the same and the setup code could probably be reused for lots of different tests.